src/ckeditor.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global CKEDITOR */
+
+import { getEditorNamespace } from './utils/geteditornamespace.js';
+
+export default {
+	name: 'ckeditor',
+
+	render( createElement ) {
+		return createElement( 'div', {}, [
+			createElement( this.tagName )
+		] );
+	},
+
+	props: {
+		value: {
+			type: String,
+			default: ''
+		},
+		type: {
+			type: String,
+			default: 'classic',
+			validator: type => [ 'classic', 'inline' ].includes( type )
+		},
+		editorUrl: {
+			type: String,
+			default: 'https://cdn.ckeditor.com/4.13.0/standard/ckeditor.js'
+		},
+		config: {
+			type: Object,
+			default: () => {}
+		},
+		tagName: {
+			type: String,
+			default: 'textarea'
+		},
+		readOnly: {
+			type: Boolean,
+			default: null // Use the null for default value, so `config.readOnly` can take precedence.
+		}
+	},
+
+	mounted() {
+		getEditorNamespace( this.editorUrl ).then( () => {
+			if ( this.$_destroyed ) {
+				return;
+			}
+
+			const config = this.config || {};
+
+			if ( this.readOnly !== null ) {
+				config.readOnly = this.readOnly;
+			}
+
+			const method = this.type === 'inline' ? 'inline' : 'replace';
+			const element = this.$el.firstElementChild;
+			const editor = this.instance = CKEDITOR[ method ]( element, config );
+
+			editor.on( 'instanceReady', () => {
+				const data = this.value;
+
+				editor.fire( 'lockSnapshot' );
+
+				editor.setData( data, { callback: () => {
+					this.$_setUpEditorEvents();
+
+					const newData = editor.getData();
+
+					// Locking snapshot prevents 'change' event.
+					// Trigger it manually to update bound data.
+					if ( data !== newData ) {
+						this.$once( 'input', () => {
+							this.$emit( 'ready', editor );
+						} );
+
+						this.$emit( 'input', newData );
+					} else {
+						this.$emit( 'ready', editor );
+					}
+
+					editor.fire( 'unlockSnapshot' );
+				} } );
+			} );
+		} );
+	},
+
+	beforeDestroy() {
+		if ( this.instance ) {
+			this.instance.destroy();
+		}
+
+		this.$_destroyed = true;
+	},
+
+	watch: {
+		value( val ) {
+			if ( this.instance.getData() !== val ) {
+				this.instance.setData( val );
+			}
+		},
+
+		readOnly( val ) {
+			this.instance.setReadOnly( val );
+		}
+	},
+
+	methods: {
+		$_setUpEditorEvents() {
+			const editor = this.instance;
+
+			editor.on( 'change', evt => {
+				const data = editor.getData();
+
+				// Editor#change event might be fired without actual data change.
+				if ( this.value !== data ) {
+					// The compatibility with the v-model and general Vue.js concept of input–like components.
+					this.$emit( 'input', data, evt, editor );
+				}
+			} );
+
+			editor.on( 'focus', evt => {
+				this.$emit( 'focus', evt, editor );
+			} );
+
+			editor.on( 'blur', evt => {
+				this.$emit( 'blur', evt, editor );
+			} );
```

Comment:
I see we don't have too many events exposed here, but maybe let's wait for feedback and see if these are needed.

Reply:
It's mostly the same set of events as for angular integration. Devs still can just use `config#on` for less common events or even access editor instance directly from the component.